### PR TITLE
opt: replace QtSingleApplication with KDSingleApplication

### PR DESCRIPTION
**Pull request by shenlebantongying**
QtSingleApplication is a small library, but there is no formal maintenance. Everyone is fixing it here and there. The effort wasted is insane. https://github.com/search?q=QtSingleApplication&type=code

Even QtCreator is patching its private copy :sweat_smile: https://github.com/qt-creator/qt-creator/tree/master/src/shared/qtsingleapplication

Just throw it away and stop wasting time.

---

KDSingleApplication also uses QLockFile but with better & modernized code. At least their code includes tests, etc. 

(The SingleApplication uses QSharedMemory, which has problems, see the issue related to Qt6.6 & M1 macOS https://github.com/itay-grudev/SingleApplication/issues)

Architecturally, an advantage of KDSingleApplication is not coupled with QApplication.

https://www.kdab.com/kdsingleapplication/
https://github.com/KDAB/KDSingleApplication

### CODE REVIEW

**Comment by shenlebantongying** [comment]
I tested this on Linux & macOS. I don't have access to windows machine at the moment.

**Comment by xiaoyifang** [comment]
Does QtSingleApplication has some defects which KDSingleApplication have solved?

any better improvements?

**Comment by shenlebantongying** [comment]
> Does QtSingleApplication has some defects which KDSingleApplication have solved?

Yes, KDSingleApplication has a company to maintain it while QtSingleApplication is abandoned. The biggest defect is not the code.

**Comment by xiaoyifang** [comment]
I think if the SingleApplication  has no issues to be fixed.  we should avoid change it right now.

Too much risk with less profit.

**Comment by shenlebantongying** [comment]
> we should avoid change it right now.

Then when would we change it :sweat_smile:? It is already done, we should change it now.

> Too much risk with less profit.

I don't see much of a risk, tt works fine with better code quality.

**Comment by xiaoyifang** [comment]
When the time arise that an issue exists in the current code which not present in the new library.

**Comment by shenlebantongying** [comment]
We should use better code without maintenance burden.

**Comment by xiaoyifang** [comment]
QSingleApplication has not been changed for a lot of years.  ..In the foreseeable near future there should be less demand for changing.
If we use the new library ,we will continue to update the code to keep update with the latest code which is not acceptable.

But if this new library has indeed made some siginificant improvements I will gladly accept it.

File: src/hotkeywrapper.cc [hunk 1]
```diff
@@ -17,27 +17,7 @@
...
-           &QGuiApplication::commitDataRequest,
-           this,
-           &QHotkeyApplication::hotkeyAppCommitData,
-           Qt::DirectConnection );
-
-  connect( this,
-           &QGuiApplication::saveStateRequest,
-           this,
-           &QHotkeyApplication::hotkeyAppSaveState,
-           Qt::DirectConnection );
```
Comment (xiaoyifang) [inline src/hotkeywrapper.cc:32]: these lines can be seperated into a seperated PR.
seems not used at all.

**Pull request closed without merging**